src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -0,0 +1,487 @@
+makes sure it gets embedded into your native app.
+3. Any plugins are added as CocoaPod pods.  In theory, it should be possible to manually merge those in as well, but those instructions
+vary on the pod dependencies of each plugin.
+4. A build script is added to the Podfile targets that call `install_all_flutter_pods` to ensure that the binaries you build stay up to date
+with the Dart code that's actually in the folder.  It also uses your Xcode build configuration (Debug, Profile, Release) to embed the matching
+release type of `Flutter.framework` and `App.framework`.
+
+## Write code to use FlutterViewController from your host app
+
+The proper place to do this will be specific to your host app. Here is an
+example that makes sense for the blank screen of the host app generated
+by Xcode.
+
+First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which help you to register a plugin without a FlutterViewController instance.
+
+In `AppDelegate.h`:
+
+```objective-c
+#import <UIKit/UIKit.h>
+#import <Flutter/Flutter.h>
+
+@interface AppDelegate : FlutterAppDelegate
+@property (nonatomic,strong) FlutterEngine *flutterEngine;
+@end
+```
+
+This allows `AppDelegate.m` to be really simple, unless your host app
+needs to override other methods here:
+
+```objective-c
+#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Only if you have Flutter Plugins
+
+#import "AppDelegate.h"
+
+@implementation AppDelegate
+
+// This override can be omitted if you do not have any Flutter Plugins.
+- (BOOL)application:(UIApplication *)application
+    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"io.flutter" project:nil];
+  [self.flutterEngine runWithEntrypoint:nil];
+  [GeneratedPluginRegistrant registerWithRegistry:self.flutterEngine];
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+```
+
+If you are writing in Swift, you can do the following in your `AppDelegate.swift`:
+
+```swift
+import UIKit
+import Flutter
+import FlutterPluginRegistrant // Only if you have Flutter Plugins.
+
+@UIApplicationMain
+class AppDelegate: FlutterAppDelegate {
+  var flutterEngine : FlutterEngine?;
+  // Only if you have Flutter plugins.
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    self.flutterEngine = FlutterEngine(name: "io.flutter", project: nil);
+    self.flutterEngine?.run(withEntrypoint: nil);
+    GeneratedPluginRegistrant.register(with: self.flutterEngine);
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions);
+  }
+
+}
+```
+
+<details>
+<summary>What to do if the app delegate already inherits from somewhere else.</summary>
+
+Make your app delegate implement the `FlutterAppLifeCycleProvider` protocol, e.g.:
```

Comment:
We don't use "e.g.". Please replace with "for example" or "such as", etc.

Reply:
done